sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/FunctionRegistry.scala

```diff
@@ -527,6 +527,7 @@ object FunctionRegistry {
     expression[MakeDate]("make_date"),
     expression[MakeTimestamp]("make_timestamp"),
     expression[MakeInterval]("make_interval"),
+    expression[MakeYMInterval]("make_ym_interval"),
```

Comment:
I think the name of interval series functions should be like following:
```
make_interval
make_duration
make_period
```
This naming convention has been adopted by Java such as [Duration](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/time/Duration.html) and [Period](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/time/Period.html).

Reply:
duration/period is not a SQL type, and users may not be able to map them to year-month and day-time intervals.

Reply:
gotcha